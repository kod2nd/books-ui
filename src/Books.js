import React, { Component } from 'react';
import { getUrlPath } from '../src/utility/getUrlPath'

class Books extends Component {
    constructor() {
        super()
        this.state = {
            books: []
        }
    }

    async getBooks() {
        const response = await fetch(getUrlPath('/books'))
        const data = await response.json()
        this.setState({
            books: data
        })
    }

    mapArray(array, attribute) {
        return array.map(element => {
            return <li key={element._id}>{element[attribute]}</li>
        })
    }

    render() {
        return (
            <div>
                <div>
                    <h2>Books</h2>
                    {this.mapArray(this.state.books, "title")}
                </div>
            </div>
        );
    }

    componentDidMount() {
        this.getBooks()
    }
}

export default Books;
