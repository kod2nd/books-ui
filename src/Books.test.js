import ShallowRenderer from 'react-test-renderer/shallow'
import React, { Component } from 'react';
import Books from './Books'

test('render Books should show a title ', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Books />)
    const output = renderer.getRenderOutput()
    expect(output).toMatchSnapshot()
});

test('renders a title and list of books after component mount ', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Books />)
    const instance = renderer.getMountedInstance()
    instance.componentDidMount()

    const output = renderer.getRenderOutput()
    expect(output).toMatchSnapshot()
});