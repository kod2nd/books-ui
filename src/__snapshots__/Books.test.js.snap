// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Books should show a title  1`] = `
<div>
  <div>
    <h2>
      Books
    </h2>
  </div>
</div>
`;

exports[`renders a title and list of books after component mount  1`] = `
<div>
  <div>
    <h2>
      Books
    </h2>
    <li>
      Harry Potter
    </li>
  </div>
</div>
`;
